### ISSUE TITLE
Remove Graphviz from Javadoc by default

### ISSUE THREAD

**Issue by morungos** [pughlab/cbioportal#13]
The Javadoc is failing due to a default dependency on Graphviz. The problematic config is below, and should really only be active when `dot` is installed wherever it is usually installed (at a wild guess `/usr/local/bin/dot`). Maven should allow this to run when a profile activates it, but only when `dot` exists. 

This ticket should refactor the Javadoc config to move this doclet into a profile that gets activated by the existence of this file. See: http://maven.apache.org/guides/introduction/introduction-to-profiles.html

````
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.9</version>
                <configuration>
                    <doclet>org.umlgraph.doclet.UmlGraphDoc</doclet>
                    <docletArtifact>
                        <groupId>org.umlgraph</groupId>
                        <artifactId>umlgraph</artifactId>
                        <version>5.6</version>
                    </docletArtifact>
                    <additionalparam>-Xdoclint:none -views -all</additionalparam>
                    <useStandardDocletOptions>true</useStandardDocletOptions>
                </configuration>
            </plugin>
```

**Comment by morungos** [comment]
And I said refactor here: the `additionalparam` and `useStandardDocletOptions` should probably apply regardless.

**Comment by morungos** [comment]
Resolved in 29c44be
